pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1294,6 +1294,10 @@ func (p *Provider) computeInstanceArgs(
 	image := providerOpts.Image
 	imageProject := defaultImageProject
 
+	if opts.Arch == string(vm.ArchARM64) && !providerOpts.useArmAMI() {
+		l.Printf("WARNING: --arch=arm64 will be ignored since --gce-machine-type is not specified")
```

Comment:
nit: should we make it a slightly more explicit hint along the lines of "use --gce-machine-type to specify a t2a VM"?

Reply:
Why don't we just bail out as done below? `useArmAMI` is false in the above case since the default machine type is amd64. Effectively, you have a mismatch between the specified architecture and machine type. 

I should also mention that the `--arch` flag was meant to be a stopgap between a machine type and the corresponding architecture. Ideally, the user wouldn't have to specify the latter [1]. Both GCE and AWS expose architecture for each machine type; for Azure, we can hardcode the mapping since machine types don't change that frequently.

[1] https://github.com/cockroachdb/cockroach/issues/105472

Reply:
@srosenberg I don't have a strong preference to any particular solution. I added a warn log as of now so that we don't fail any ongoing behaviour that uses the current behaviour.
If the majority reviewers are in favour of bailing out in this case, I'm fine with returning instead of adding a warn log.

Reply:
@srosenberg @DarrylWong I have updated the change to error out if this case happens. PTAL